### ISSUE TITLE
Iterate on TypeScript CI

### ISSUE THREAD

**Issue by timsuchanek** [prisma/prisma#3004]
As of now, we run our TypeScript CI tests in Buildkite due to historical reasons.
The current setup, however, has a few limitations:
- [x] Gets quite slow - 12min to publish. Tests should run in parallel
- [x] We don't test Mac or Windows or any older Node.js versions
- [ ] Updating Snapshots takes a lot of time, because it has to be done locally
- [ ] Decouple new engine builds from the main TS CI - a new engine should not block the current TS CI
- [x] The whole CI stops on the first error - instead, we should see errors of  all packages
- [x] Prisma Client types tests should use test.serial
- [x] Fix `prisma doctor` test
- [x] Cleanup verbose pnpm logging
- [ ] Don't build packages multiple times
- [x] Integration tests in the CLI package should be migrated from Mocha to Jest

**Comment by Jolg42** [comment]
https://github.com/prisma/prisma/issues/3553

**Comment by matthewmueller** [comment]
Closing in favor of more concrete issues.

**Comment by timsuchanek** [comment]
Most of the work is happening here https://github.com/prisma/prisma/issues/3723
